vernac/vernacentries.ml

```diff
@@ -1605,9 +1617,9 @@ let vernac_set_append_option ~locality key s =
 
 let vernac_set_option ~locality table v = match v with
 | OptionSetString s ->
-  (* We make a special case for warnings because appending is their
-  natural semantics *)
-  if CString.List.equal table ["Warnings"] then
+  (* We make a special case for warnings and debug flags because appending is
+  their natural semantics *)
+  if CString.List.equal table ["Warnings"] || CString.List.equal table ["Debug"] then
     vernac_set_append_option ~locality table s
```

Comment:
Didn't  we want to move away from the implicit Append of Set Warnings at some point?

Reply:
I don't think so, what for?

Reply:
There were issues discussed in [#305](https://github.com/coq/coq/pull/305#issuecomment-252095355). I did not compare to the current situation, so I don't know if there are still relevant.

Reply:
>  I did not compare to the current situation, so I don't know if there are still relevant.

Just checking now: the issues are solved thanks to a "normalization" on the fly of the string (`normalize_flags_string`), so that the string correctly encodes a map, i.e. so that the append correctly simulates a `Map.add`.

Reply:
It is very likely that what I did here needs to be refined similarly, indeed. However, I wanted to gather a bit of feedback about the feature first.